src/http_server.cc

```diff
@@ -3729,25 +3729,19 @@ HTTPAPIServer::InferRequestClass::InferResponseComplete(
   HTTPAPIServer::InferRequestClass* infer_request =
       reinterpret_cast<HTTPAPIServer::InferRequestClass*>(userp);
 
-  auto response_count = infer_request->IncrementResponseCount();
-
-  // Defer to the callback with the final response
-  if ((flags & TRITONSERVER_RESPONSE_COMPLETE_FINAL) == 0) {
-    LOG_ERROR << "[INTERNAL] received a response without FINAL flag";
-    return;
+  if (response != nullptr) {
+    ++infer_request->response_count_;
   }
 
   TRITONSERVER_Error* err = nullptr;
-  if (response_count != 0) {
-    err = TRITONSERVER_ErrorNew(
-        TRITONSERVER_ERROR_INTERNAL, std::string(
-                                         "expected a single response, got " +
-                                         std::to_string(response_count + 1))
-                                         .c_str());
-  } else if (response == nullptr) {
+  if (infer_request->response_count_ != 1) {
     err = TRITONSERVER_ErrorNew(
-        TRITONSERVER_ERROR_INTERNAL, "received an unexpected null response");
-  } else {
+        TRITONSERVER_ERROR_INTERNAL,
+        std::string(
+            "expected a single response, got " +
+            std::to_string(infer_request->response_count_))
+            .c_str());
+  } else if (response != nullptr) {
```

Comment:
For the case:
```
TRITONBACKEND_ResponseSend(response, 0, nullptr /* success */);
TRITONBACKEND_ResponseFactorySendFlags(factory, TRITONSERVER_RESPONSE_COMPLETE_FINAL);
```

Am I correctly understanding that the "INFER_RESPONSE_COMPLETE" trace would get captured twice? I think for opentelemetry this is an "append" operation rather than an "overwrite". CC @oandreeva-nv

Reply:
it is an "append" operation. Need to verify this logic and will get back

Reply:
Seems like it, both of them call for `response_fn` callback, which captures "INFER_RESPONSE_COMPLETE" timestamp. I don't think this is that big of a deal for this PR. If we want to address it, file a jira ticket.

Reply:
Alternatively, this behavior can be described in docs

Reply:
Actually, IIUC 
`TRITONBACKEND_ResponseSend(response, 0, nullptr /* success */);` sets `flags` to 0, so
`flags & TRITONSERVER_RESPONSE_COMPLETE_FINAL` will be 0, and per [line](https://github.com/triton-inference-server/server/pull/6017/files#diff-2d773636d85b98207338699c3cc2b9fc7d31f38e45131ca127dca73f45d4bbb2R3759), timestap capture should be skipped. Is this understanding correct?

Reply:
Yes, in this case we expect only one response so we could just record the response complete time when we see final signal